### ISSUE TITLE
Should we automatically define some variables?

### ISSUE THREAD

**Issue by justinsb** [kopeio/kexpand#10]
e.g. timestamp or gitsha?

**Comment by dennis-bsi** [comment]
I have a use for `basename`.  Our definition files for deployments resembles

**Comment by denniswebb** [comment]
I would actually like something like this for `basename $PWD`.  Our typical layout of definitions looks like this:

```
apps
├── app1
│   ├── configmap.tmpl.yaml
│   ├── deployment.tmpl.yaml
│   ├── ingress.tmpl.yaml
│   ├── service.tmpl.yaml
│   └── values.yaml
├── app2
│   ├── configmap.tmpl.yaml
│   ├── deployment.tmpl.yaml
│   ├── ingress.tmpl.yaml
│   ├── service.tmpl.yaml
│   └── values.yaml
└── app3
    ├── configmap.tmpl.yaml
    ├── deployment.tmpl.yaml
    ├── ingress.tmpl.yaml
    ├── service.tmpl.yaml
    └── values.yaml
```

Typically each template file has a variable name `app` that is replaced with the name of the application defined in `values.yaml`.  This application name is the same as the base directory name, e.g. `app: app1`.  If I could simply define `basename` in my template, it would be 1 less variable to define a value for.

I can also see the use in `timestamp`.

I have created a prototype #11 of an implementation of default values.  These would be defined first, and could be overwritten by `-k` values of values in var files.  Below is a sample of the output.

```
~/kex $ cat values.yaml
name: Dennis
~/kex $ cat template.yaml
apiVersion: v1
kind: Secret
metadata:
  name: $((basename))
data:
  quotedName: $(name)
  unixdate: {{unixdate}}
  unixtime: $((unixtime))
  ansicdateutc: $((ansicdateutc))
  gitsha: $((gitsha))
  gitsha: $((gitshashort))
~/kex $ kexpand expand -f values.yaml template.yaml
apiVersion: v1
kind: Secret
metadata:
  name: kex
data:
  quotedName: "Dennis"
  unixdate: Sat Nov  5 00:00:12 CDT 2016
  unixtime: 1478322012
  ansicdateutc: Sat Nov  5 05:00:12 2016
  gitsha: af19fb660530323e2c54828afa6b71c698434333
  gitsha: af19fb6
```

**Comment by justinsb** [comment]
Merged #11 which added this - thank you @denniswebb
